mtcpclient/groupofconnections.go

```diff
@@ -46,6 +55,60 @@ func (gc GroupOfConnections) AtLeastOneConnectionInError() bool {
 	return gc.isIn(tcpclient.ConnectionError)
 }
 
-func (gc GroupOfConnections) AtLeastOneConnectionEstablished() bool {
-	return gc.isIn(tcpclient.ConnectionEstablished)
+func (gc GroupOfConnections) AtLeastOneConnectionOK() bool {
+	return gc.isIn(tcpclient.ConnectionEstablished) || gc.isIn(tcpclient.ConnectionClosed)
+}
+
+func appendConnections(previous_connections []tcpclient.Connection, new_connections ...tcpclient.Connection) []tcpclient.Connection {
+	// TODO implement as a method using as a reference
+	m := len(previous_connections)
+	n := m + len(new_connections)
+	if n > cap(previous_connections) {
+		// allocate double what's needed, for future growth.
+		newSlice := make([]tcpclient.Connection, (n+1)*2)
+		copy(newSlice, previous_connections)
+		previous_connections = newSlice
+	}
+	previous_connections = previous_connections[0:n]
+	copy(previous_connections[m:n], new_connections)
+	return previous_connections
+}
+
+func (gc GroupOfConnections) getFilteredListByStatus(statuses []tcpclient.ConnectionStatus) (filteredConnections []tcpclient.Connection) {
```

Comment:
as happens previously... why we are not returning a group of connections? This would also allow us to concatenate functions (example: gc.filter(A).sumThem())